### DISCUSSION TITLE
[Projects beta] GitHub App access

### DISCUSSION THREAD

**Discussion by jahredhope** [community/community#5939] [Projects and Issues]
I think GitHub Apps cannot be given access to the new Projects beta GraphQL api mutations.

Being able to use a GitHub App is important for me as it allows me to limit the access of the automation tool and easily listen to webhooks.

Am I correct that GitHub Apps can not be given access to the new Project's beta and (presumably) this functionality will be coming sometime in the future?

Thanks

---

### Details

I recieve the following error when attempting to add an item to Project beta board using a GitHub app.

**Error:**
```
GraphqlResponseError: Request failed due to following response errors:
     - Resource not accessible by integration
```

This is the query I used, which worked with a Personal Access Token.

**Mutation:**
```gql
mutation ($projectId: ID!, $contentId: ID!) {
    addProjectNextItem(input: {projectId: $projectId, contentId: $contentId}) {
      projectNextItem {
        id
    }
  }
}
```

**App permissions:**
Issue: read
Repository: Project Read/Write
Organisation: Project Read/Write

**Comment by jtoar** [comment]
I just ran into this as well. Any update on this? I know projects beta has an "automations" UI that's coming, but I wanted to implement automation sooner than that.

**Comment by compso** [comment]
What is the recommended method in the meantime?

I note that you are recommending the use of Installation Access Tokens for Github App in the projects (Beta) documentation: https://docs.github.com/en/issues/trying-out-the-new-projects-experience/using-the-api-to-manage-projects

**Reply by compso** [reply]
Never mind this seems to be working just needed to make sure the permissions were set correctly and accepted by the app installation
